nifi-nar-bundles/nifi-elasticsearch-bundle/nifi-elasticsearch-processors/src/main/java/org/apache/nifi/processors/elasticsearch/PutElasticsearchHttp.java

```diff
@@ -356,6 +363,7 @@ public void onTrigger(final ProcessContext context, final ProcessSession session
                                             errorReason = reason;
                                             logger.error("Failed to process {} due to {}, transferring to failure",
                                                     new Object[]{flowFile, errorReason});
+                                            flowFile = session.putAttribute(flowFile, "es.failure.reason", errorReason);
```

Comment:
Reviewing this line in comparison to line 369, is there any difference between the `es.failure.reason` and `reason` attributes?  If the values are the same in practice, then perhaps just updating the WritesAttribute annotation to add the `reason` attribute would be helpful.

As listed in the PR checklist, introducing a unit test to evaluate the presence of the attribute on failure would also be useful.